openeo/extra/job_management/__init__.py

```diff
@@ -877,6 +967,52 @@ def _merge_into_df(self, df: pd.DataFrame):
         else:
             self._df = df
 
+    def _update_row(self, job_id: str, updates: dict):
```

Comment:
It's not ideal to introduce a new required API `_update_row` here, while there is already `persist`, which is very similar, e.g. see it's docs:
>  Store job data to the database.
>  The provided dataframe may contain partial information, which is merged into the larger database.